indexer/centers_table.cpp

```diff
@@ -0,0 +1,243 @@
+#include "indexer/centers_table.hpp"
+
+#include "indexer/coding_params.hpp"
+#include "indexer/geometry_coding.hpp"
+#include "indexer/point_to_int64.hpp"
+
+#include "coding/memory_region.hpp"
+#include "coding/reader.hpp"
+#include "coding/succinct_mapper.hpp"
+#include "coding/varint.hpp"
+#include "coding/writer.hpp"
+
+#include "base/assert.hpp"
+
+#include "std/unordered_map.hpp"
+
+#include "3party/succinct/elias_fano.hpp"
+#include "3party/succinct/rs_bit_vector.hpp"
+
+namespace search
+{
+namespace
+{
+class CentersTableV0 : public CentersTable
+{
+public:
+  static const uint32_t kFrequency = 64;
+
+  struct Header
+  {
+    bool IsValid() const
+    {
+      if (m_base.m_version != 0)
+        return false;
+      if (m_positionsOffset < m_identifiersOffset)
+        return false;
+      if (m_deltasOffset < m_positionsOffset)
+        return false;
+      if (m_endOffset < m_deltasOffset)
+        return false;
+      return true;
+    }
+
+    CentersTable::Header m_base;
+    uint32_t m_identifiersOffset = 0;
+    uint32_t m_positionsOffset = 0;
+    uint32_t m_deltasOffset = 0;
+    uint32_t m_endOffset = 0;
+    char m_padding[4] = {};
+  };
+
+  static_assert(sizeof(Header) == 24, "Wrong header size.");
+
+  CentersTableV0(Reader & reader, serial::CodingParams const & codingParams)
+    : m_reader(reader), m_codingParams(codingParams)
+  {
+  }
+
+  // CentersTable overrides:
+  bool Get(uint32_t id, m2::PointD & center) override
+  {
+    if (id > m_ids.size() || !m_ids[id])
```

Comment:
Bad things happens. assert in debug, ud in release. Added regression test, fixed, thanks!